clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4707,6 +4711,35 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
     O << "  __SYCL_DLL_LOCAL\n";
     O << "  static constexpr bool isESIMD() { return " << K.IsESIMDKernel
       << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFileName() { return "
+#ifndef NDEBUG
+      << std::string(PLoc.getFilename())
+             .substr(std::string(PLoc.getFilename()).find_last_of("/\\") + 1)
+#endif
+      << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFunctionName() { return "
+#ifndef NDEBUG
+      << K.NameType->getAsCXXRecordDecl()->getName()
```

Comment:
Alexey commented on the #ifndef lines, you absolutely don't want those in SemaSYCL.cpp (they are to be printed presumably?).

Is this really sufficient to meet the requirements to just print the provided name?  I would expect the users would want a fully-qualified name.

Reply:
@alexbatashev Just checking: I am working to emit a fully-qualified name. Please let me know if that is not required.

Reply:
Pushed https://github.com/intel/llvm/pull/5335/commits/2b10cbfd14da03df5e16a8ce3de21e6f3594539c

Reply:
@schittir there's no strict requirements here as long as users can understand the name and where it originates from.

Reply:
I would still think FQN is much more valuable (and by your statement, likely _IS_ a requirement :D).  Consider:

namespace A {
void foo() {
kernel<class KernelName>(...);
}
namespace B {
void foo() {
kernel <class KernelName>(...);
}
}

Without the FQN, you'd get "KernelName" for both.  Additionally, in the 'unnamed lambda' case, consider that EVERY lambda ends up having the name "operator()" without it being the FQN.